src/command_modules/azure-cli-role/tests/test_role.py

```diff
@@ -8,73 +8,112 @@
-        self.assertEqual(self.sp_name, result['name'])
+        result = self.cmd('ad sp reset-credentials -n {0} --create-cert'.format(sp_name)).get_output_in_json()
+        self.assertEqual(sp_name, result['name'])
         self.assertTrue(result['fileWithCertAndPrivateKey'].endswith('.pem'))
         os.remove(result['fileWithCertAndPrivateKey'])
+        self.cmd('ad app delete --id {}'.format(sp_name), checks=NoneCheck())
+
+
+class RbacSPKeyVaultScenarioTest(LiveTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_sp_with_kv_new_cert')
+    @KeyVaultPreparer()
+    def test_create_for_rbac_with_new_kv_cert(self, resource_group, key_vault):
+
+        import time
+        sp_name = 'http://{}'.format(resource_group)
+
+        subscription_id = self.cmd('account list --query "[?isDefault].id"').get_output_in_json()
+        scope = '/subscriptions/{}'.format(subscription_id[0])
+        cert_name = 'cert1'
+        time.sleep(5)
 
-        self.cmd('ad app delete --id {}'.format(self.sp_name), checks=NoneCheck())
+        self.cmd('ad sp create-for-rbac --scopes {0} {0}/resourceGroups/{1} --create-cert --keyvault {2} --cert {3} -n {4}'.format(
+            scope, resource_group, key_vault, cert_name, sp_name)).get_output_in_json()
+        self.cmd('ad sp reset-credentials -n {0} --create-cert --keyvault {1} --cert {2}'.format(sp_name, key_vault, cert_name))
+        self.cmd('ad app delete --id {}'.format(sp_name))
+
+    @ResourceGroupPreparer(name_prefix='cli_test_sp_with_kv_existing_cert')
+    @KeyVaultPreparer()
+    def test_create_for_rbac_with_existing_kv_cert(self, resource_group, key_vault):
+
+        import time
+        sp_name = 'http://{}'.format(resource_group)
+
+        subscription_id = self.cmd('account list --query "[?isDefault].id"').get_output_in_json()
+        scope = '/subscriptions/{}'.format(subscription_id[0])
+        cert_name = 'cert1'
+        time.sleep(5)
+
+        # test with valid length cert
+        policy = self.cmd('keyvault certificate get-default-policy').get_output_in_json()
+        self.cmd('keyvault certificate create --vault-name {} -n {} -p "{}" --validity 24'.format(key_vault, cert_name, policy))
+        self.cmd('ad sp create-for-rbac --scopes {0} {0}/resourceGroups/{1} --keyvault {2} --cert {3} -n {4}'.format(
+            scope, resource_group, key_vault, cert_name, sp_name)).get_output_in_json()
+        self.cmd('ad sp reset-credentials -n {0} --keyvault {1} --cert {2}'.format(sp_name, key_vault, cert_name))
+        self.cmd('ad app delete --id {}'.format(sp_name))
+
+        # test with cert that has too short a validity
+        sp_name = '{}2'.format(sp_name)
+        self.cmd('keyvault certificate create --vault-name {} -n {} -p "{}" --validity 6'.format(key_vault, cert_name, policy))
+        self.cmd('ad sp create-for-rbac --scopes {0} {0}/resourceGroups/{1} --keyvault {2} --cert {3} -n {4}'.format(
```

Comment:
We don't check for the warning but if the desired behavior didn't trigger, then this create would fail with the "bad key date" error.